lib/airbrake-ruby.rb

```diff
@@ -275,20 +275,19 @@ def create_deploy(deploy_params, notifier = :default)
     private
 
     ##
-    # Calls +method+ on +notifier+ with provided +args+.
+    # Calls +method+ on +notifier+ with provided +args+. If +notifier+ is not
+    # configured, returns an error hash.
     #
-    # @raise [Airbrake::Error] if none of the notifiers exist
+    # @return [Hash{String=>String}, Object]
     def call_notifier(notifier, method, *args, &block)
-      if @notifiers.key?(notifier)
-        @notifiers[notifier].__send__(method, *args, &block)
-      else
-        # If we raise SystemExit, the Ruby process can gracefully quit without
-        # the unwanted Airbrake::Error.
-        raise args.first if args.first.class == SystemExit
-
-        raise Airbrake::Error,
-              "the '#{notifier}' notifier isn't configured"
+      unless @notifiers.key?(notifier)
+        return {
+          'error' => "#{LOG_LABEL} the '#{notifier}' notifier isn't configured, " \
+                     "'#{method}' was not called"
+        }
```

Comment:
This looks wrong to me, because now every notifier method can return hash `{error: "qqqq"}` (even method that return bool etc) and that is a bad design.